### PR TITLE
pkg/profilecreator: Ensure targeted nodes have the same hardware

### PR DESCRIPTION
**Pull request by marcel-apf**
While at it:
 - Update the unit tests to use a const must-gather dir
 - Fix nodes association to MCPs
 - Add unit-tests for the node equality
 - Fix the wrapper
 - Improve must-gather dir browsing
 - Default split-reserved-cpus-across-numa to false
 - Use csvtools to create the profile - results in an usable yaml file
 - Add the node-selector to profile - results in an usable profile
 - Update dependencies …

### CODE REVIEW

File: pkg/profilecreator/profilecreator_test.go [hunk 1]
```diff
@@ -10,6 +10,10 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
```
Comment (swatisehgal) [inline pkg/profilecreator/profilecreator_test.go:17]: This file has merge conflicts, we need to rebase.

**Review by swatisehgal — CHANGES_REQUESTED** [review]
Just a minor suggestion to squash two commits. We need to rebase too but other than that looks good!

File: pkg/profilecreator/profilecreator_test.go [hunk 2]
```diff
@@ -142,7 +142,7 @@ var _ = Describe("PerformanceProfileCreator: Consuming GHW Snapshot from Must Ga
 })
 
 var _ = Describe("PerformanceProfileCreator: Populating Reserved and Isolated CPUs in Performance Profile", func() {
-	var mustGatherDirPath, mustGatherDirAbsolutePath string
+	var mustGatherDirAbsolutePath string
```
[the remark this answers is not in the document]
Comment (marcel-apf) [reply pkg/profilecreator/profilecreator_test.go:147]: Sure, thanks. I rebased  once and I had to add on top.

File: pkg/profilecreator/profilecreator_test.go (hunk 1 above)
Comment (marcel-apf) [reply pkg/profilecreator/profilecreator_test.go:16]: Done

**Pull request merged**